supportedBackends/aws/src/main/resources/agent/launch_sidecar

```diff
@@ -0,0 +1,92 @@
+#!/bin/sh
+# Copyright 2018 Amazon.com, Inc. or its affiliates.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#
+#  1. Redistributions of source code must retain the above copyright notice,
+#  this list of conditions and the following disclaimer.
+#
+#  2. Redistributions in binary form must reproduce the above copyright
+#  notice, this list of conditions and the following disclaimer in the
+#  documentation and/or other materials provided with the distribution.
+#
+#  3. Neither the name of the copyright holder nor the names of its
+#  contributors may be used to endorse or promote products derived from
+#  this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+#  BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+#  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+#  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+#  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+#  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+#  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+#  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+#  IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+
+function environmentVariable {
+  query=".[0].Config.Env[] | select(startswith(\""${1}"\"))"
+  echo $containerInfo |jq -r "$query" | cut -d= -f2
+}
+
+# $1 is passed in by spawnoninput, which originally will come from a single
+# line from the docker event stream (endpoint /events)
+line=$1
+
+# These commands are for "start", "create", "attach"
+# While jq is functionally preferred, cut should be slightly faster. We're
+# somewhat against the clock here as the container is going to do some work
+dockerid=$(echo "$1" | cut -d: -f3 | cut -d\" -f2)
+#dockerid=$(echo "$1" | jq -r '.id')
+echo 'pausing id: ' "$dockerid"
+docker pause $dockerid
+# Docker pause can happen based on the "create" event from docker
+#
+# The infrastructure around this seems to take between 10 and 20ms in my tests.
+# This is likely due to the processing time needed for jq to parse the json.
+#
+# It is possible, though unlikely, that some dependent work starts happening
+# within that time. If so, we may need a sleep 0.02 in the script preamble
+# to accomodate this scenario and/or find something more performant than jq.
+# Because of this, we are using cut above rather than jq, which should be slightly faster
```

Comment:
Have you measured how multiple jobs being scheduled in parallel on the same host would affect the sidecar performance ?

Reply:
The biggest risk on sidecar performance is in regards to the limitations of inotify, especially wrt large directory trees with lots of small files. But then, that's not an ideal situation for s3 either. I haven't done much measurement of these limitations, but I don't expect number of containers to be an issue as much as the aggregate number of files involved.

Reply:
Oh I see, when I said sidecar I actually meant the agent. But if all it does is spawning a new process it should not be too much of a bottleneck.
As for the actual sidecar, how will it know what files to localize and delocalize ? For delocalization you seem to imply it will watch the filesystem and push everything to S3 ?

Reply:
The environment variables I pass through to the task (the AWS_CROMWELL stuff) have the necessary information the sidecar needs to localize/delocalize. I'll have a new PR for that shortly.